### DISCUSSION TITLE
Modbus Communication Error_ Yoshan 2kg

### DISCUSSION THREAD

**Discussion by gnosiscoffeeroasters** [artisan-roaster-scope/artisan#905] [Q&A] [answered]
I have a brand new Yoshan 2k, using PC.  I have installed the Windows Driver sent by Yoshan, and am using the USB cable they sent me. 

When I click "On" and "Start" on Artisan I get a Modbus Communication Error and there are no temperature readings. Any ideas?

**Comment by poundy** [comment]
The first thing with a new roaster connected to a P, when it works over Serial connection, is to make sure you selected the correct COM port in the machine setup. How did you determine the correct one to use?  The simplest way I always fall back to is to disconnect the roaster, open Device Manager on your Windows PC, and you should see any serial ports you have by default (often none, and the "Ports (COM & LPT)" section isn't visible). Note any COM numbers that may be evident. Then, with Device Manager still open, plug in the roaster; this should refresh the view in Device Manager and you should see the COM port appear. That will tell you two things - first, what the COMx number needs to be in Artisan so the software can talk to the hardware, and second, that the driver has loaded correctly (no exclamation mark against it). Once the driver loads correctly, and you have Artisan set to use the correct number, try again. This then eliminates the setup as a potential source of error. Report back on if that helps !

**Comment by gnosiscoffeeroasters** [comment, marked as answer]
I figured it out. I needed to install the Universal Windows Driver, instead of the old version that I accidentally installed.
